src/allotropy/parsers/thermo_skanit/thermo_skanit_structure.py

```diff
@@ -0,0 +1,277 @@
+        )
+        msg = (
+            "Unable to identify plate well count from Layout definitions tab. "
+            "Expected to find in Plate template description."
+        )
+        if plate_template_value is None:
+            raise AllotropyParserError(msg)
+        else:
+            # Extract the integer from the combined string
+            search_string = re.search(r"\d+", plate_template_value)
+            if search_string:
+                plate_template_number = int(search_string.group())
+                return plate_template_number
+            else:
+                raise AllotropyParserError(msg)
+
+    @staticmethod
+    def identify_abs_and_sample_dfs(
+        absorbance_sheet_df: pd.DataFrame,
+    ) -> tuple[pd.DataFrame, pd.DataFrame, float]:
+        # Initialize variables
+        start_reading_abs = False
+        start_reading_sample = False
+        data_between_abs_and_blank = []
+        data_between_sample_and_blank = []
+
+        # Iterate through each row
+        for _index, row in absorbance_sheet_df.iterrows():
+            if pd.notna(row.iloc[0]) and "Wavelength" in row.iloc[0]:
+                match = re.search(r"Wavelength:\s*(\d{3})\s*nm", row.iloc[0])
+                if match:
+                    wavelength = float(match.group(1))
+                else:
+                    msg = "Unable to identify Wavelength (nm) from Absorbance tab"
+                    raise AllotropyParserError(msg)
+            if start_reading_sample:
+                # Check if the row is blank
+                if row.isnull().all():
+                    break
+                data_between_sample_and_blank.append(row)
+            elif start_reading_abs:
+                # Check if the row is blank
+                if row.isnull().all():
+                    start_reading_abs = False
+                    start_reading_sample = False  # Reset this to prevent starting the next DataFrame before finding 'sample'
+                else:
+                    data_between_abs_and_blank.append(row)
+            elif "Abs" in row.values.astype(str):  # Check if 'abs' is in the row
+                start_reading_abs = True
+            elif "Sample" in row.values.astype(str):  # Check if 'sample' is in the row
+                start_reading_sample = True
+
+        # Create DataFrames with the collected data
+        df_between_abs_and_blank = pd.DataFrame(data_between_abs_and_blank).reset_index(
+            drop=True
+        )
+        df_between_sample_and_blank = pd.DataFrame(
+            data_between_sample_and_blank
+        ).reset_index(drop=True)
+
+        return (df_between_abs_and_blank, df_between_sample_and_blank, wavelength)
+
+
+@dataclass(frozen=True)
+class DataThermoSkanIt(Data):
+    @staticmethod
+    def create(sheet_data: dict[str, pd.DataFrame], file_name: str) -> Data:
+        for sheet_name, df in sheet_data.items():
+            # NOTE: This assumes a single absorbance plate on a single tab
```

Comment:
From mentions above, if we clean the dataframes by dropping empty columns, we can read the data more easily:

```
def _clean_dataframe(df: pd.DataFrame):
    df = df.replace(r"^\s*$", np.nan, regex=True)
    df = df.dropna(axis="index", how="all")
    df = df.dropna(axis="columns", how="all")
    return df
```
```
        for sheet_name, df in sheet_data.items():
            df = _clean_dataframe(df)
```